doc/md/concepts/workflows.md

```diff
@@ -0,0 +1,139 @@
+---
+title: Declarative vs Versioned Workflows
+id: workflows
+slug: /concepts/declarative-vs-versioned
+---
+
+This section introduces two types of workflows that are supported by Atlas
+to manage database schemas _declarative_ and _versioned_ migrations. 
+
+### Declarative Migrations
+
+The declarative approach has become so popular with engineers nowadays because it embodies
+a convenient separation of concerns between application and infrastructure engineers.
+Application engineers describe _what_ (the desired state) they need to happen, and
+infrastructure engineers build tools that plan and execute ways to get to that state (_how_).
+This division of labor allows for great efficiencies as it abstracts away the complicated
+inner workings of infrastructure behind a simple, easy to understand API for the application
+developers and allows for specialization and development of expertise to pay off for the
+infra people.
+
+With declarative migrations, the desired state of the database schema is given 
+as input to the migration engine, which plans and executes a set of actions to
+change the database to its desired state.
+
+For example, suppose your application uses a small SQLite database to store its data.
+In this database, you have a `users` table with this structure:
+```hcl
+schema "main" {}
+
+table "users" {
+  schema = schema.main
+  column "id" {
+    type = int
+  }
+  column "greeting" {
+    type = text
+  }
+}
+```
+Now, suppose that you want to add a default value of `"shalom"` to the `greeting`
+column. Many developers are not aware that it isn't possible to modify a column's
+default value in an existing table in SQLite. Instead, the common practice is to
+create a new table, copy the existing rows into the new table and drop the old one
+after. Using the declarative approach, developers can change the default value for
+the `greeting` column:
+
+```hcl {10}
+schema "main" {}
+
+table "users" {
+  schema = schema.main
+  column "id" {
+    type = int
+  }
+  column "greeting" {
+    type = text
+    default = "shalom"
+  }
+}
+```
+And have Atlas's engine devise a plan similar to this:
+```sql
+-- Planned Changes:
+-- Create "new_users" table
+CREATE TABLE `new_users` (`id` int NOT NULL, `greeting` text NOT NULL DEFAULT 'shalom')
+-- Copy rows from old table "users" to new temporary table "new_users"
+INSERT INTO `new_users` (`id`, `greeting`) SELECT `id`, IFNULL(`greeting`, ?) AS `greeting` FROM `users`
+-- Drop "users" table after copying rows
+DROP TABLE `users`
+-- Rename temporary table "new_users" to "users"
+ALTER TABLE `new_users` RENAME TO `users`
+```
+
+### Versioned Migrations
+
+As the database is one of the most critical components in any system, applying changes
+to its schema is rightfully considered a dangerous operation. For this reason, many teams
```

Comment:
Consider mentioning another reason, where one schema is deployed on many environments that are not accessible.